mcs/build/profiles/mobile.make

```diff
@@ -16,7 +16,6 @@ DEFAULT_REFERENCES = -r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll
 PROFILE_MCS_FLAGS = \
 	-d:NET_1_1 \
 	-d:NET_2_0 \
-	-d:NET_2_1 \
```

Comment:
I have no strong opinion on removing NET_2_1 in this PR. But for the sake of having as little breaking change as possible, I will leave the removing of it for another PR.
In the same note, but for yet another PR, I am pretty sure we could get rid of NET_1_1, NET_2_0 and NET_3_5 too.